### PR TITLE
Add local cache used pages size metric

### PR DESCRIPTION
**Pull request by bzheng888**
### What changes are proposed in this pull request?
Add local cache pages used metric 

### Why are the changes needed?
Sometimes we not only need space used information, but also the pages size information. The pages size give us meta information for future analyze.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 1]
```diff
@@ -1417,6 +1417,12 @@ public MetricKey build() {
           .setMetricType(MetricType.GAUGE)
           .setIsClusterAggregated(false)
           .build();
+  public static final MetricKey CLIENT_CACHE_PAGE_SIZE =
+      new Builder("Client.CachePageSize")
+          .setDescription("Amount of pages used by the client cache.")
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:1422]: As the description you gave, do you think the metric name `CLIENT_CACHE_PAGE_COUNT` can make sense more?

**Review by maobaolong — CHANGES_REQUESTED** [review]
Thanks for this metrics export, Left one comment inline.

**Review by maobaolong — APPROVED** [review]
LGTM

**Comment by maobaolong** [comment]
@beinan Would you please take another look?

**Comment by maobaolong** [comment]
@beinan ping

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 2]
```diff
@@ -722,6 +722,9 @@ private static void registerGauges(long cacheSize, MetaStore metaStore) {
       MetricsSystem.registerGaugeIfAbsent(
           MetricsSystem.getMetricName(MetricKey.CLIENT_CACHE_SPACE_USED.getName()),
           metaStore::bytes);
+      MetricsSystem.registerGaugeIfAbsent(
+          MetricsSystem.getMetricName(MetricKey.CLIENT_CACHE_PAGE_COUNT.getName()),
```
Comment (beinan) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:726]: I think there has been a metrics key called client_cache_pages.    https://github.com/Alluxio/alluxio/blob/a10823a9523e200f1665228cd4eb3ea7659a0d15/core/common/src/main/java/alluxio/metrics/MetricKey.java#L1488

Comment (beinan) [reply core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java:726]: If you just wanna count the total number of the pages in our local cache, why not just use this metrics?

https://github.com/Alluxio/alluxio/blob/f2825833fbf5d473fe7265aae749e8b3d52df143/core/client/fs/src/main/java/alluxio/client/file/cache/DefaultMetaStore.java#L146

https://github.com/Alluxio/alluxio/blob/f2825833fbf5d473fe7265aae749e8b3d52df143/core/client/fs/src/main/java/alluxio/client/file/cache/DefaultMetaStore.java#L71

**Comment by maobaolong** [comment]
Hi @beinan , we have some bad experience for the counter which counted by increase and decrease, it maybe wrong when we encounter exception, how about use `mPageMap.size()` instead of the `mPages` and `Metrics.PAGES`

**Comment by beinan** [comment]
> Hi @beinan , we have some bad experience for the counter which counted by increase and decrease, it maybe wrong when we encounter exception, how about use `mPageMap.size()` instead of the `mPages` and `Metrics.PAGES`

Good call, create https://github.com/Alluxio/alluxio/pull/15121 to use the actual size of the map instead of the mPages

**Comment by beinan** [comment]
hello @bzheng888 & @maobaolong , since the metrics of num of pages has been there already, can I shutdown this PR?

**Comment by bzheng888** [comment]
@beinan LGTM! Thanks!

**Pull request closed without merging**